### PR TITLE
Improvements proposition

### LINKED ISSUE #26: Support `[YANKED]` tag
**Is your feature request related to a problem? Please describe.**

Keep a changelog define that versions that has been pulled out for security reason should be tagged with `yanked`

CleanShot 2023-12-14 at 21 43 29@2x

We should make KeepAChangelog aware of that tag.

### PR DESCRIPTION
**Pull request by MangelMaxime**
### WHAT
copilot:summary

copilot:poem

copilot:emoji

### HOW
copilot:walkthrough

Sorry for the big PR, the auto formatting didn't help on that here 🙈

I think copilot for PR is not available anymore so I will write a good old description myself ahah.

- Change `Releases` from a list of tuple to using a list of record has we are starting to have a lot of properties
- Disable formatting on save as it makes too many changes and #24 addresses the formatting setup
- Fix #26 

I was not able to test the `Ionide.KeepAChangelog.Tasks` project because when packing it there seems to be missing some `dll`. 

I think this is something that was discovered with the last release already

```
➜  Thoth.Json.Core git:(feature/kevin_pr) ✗ dotnet pack
MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
  Restored /Users/mmangel/Workspaces/Github/thoth-org/Thoth.Json/packages/Thoth.Json.Core/Thoth.Json.Core.fsproj (in 800 ms).
/Users/mmangel/.nuget/packages/ionide.keepachangelog.tasks/900.2.0/build/Ionide.KeepAChangelog.Tasks.targets(29,9): error MSB4062: The "Ionide.KeepAChangelog.Tasks.ParseChangeLogs" task could not be loaded from the assembly /Users/mmangel/.nuget/packages/ionide.keepachangelog.tasks/900.2.0/build/../tasks/net6.0/Ionide.KeepAChangelog.Tasks.dll. Could not load file or assembly 'FSharp.Core, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. [/Users/mmangel/Workspaces/Github/thoth-org/Thoth.Json/packages/Thoth.Json.Core/Thoth.Json.Core.fsproj]
/Users/mmangel/.nuget/packages/ionide.keepachangelog.tasks/900.2.0/build/Ionide.KeepAChangelog.Tasks.targets(29,9): error MSB4062:  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [/Users/mmangel/Workspaces/Github/thoth-org/Thoth.Json/packages/Thoth.Json.Core/Thoth.Json.Core.fsproj]
```

### CODE REVIEW

File: src/Ionide.KeepAChangelog/Library.fs [hunk 1]
```diff
@@ -238,32 +226,60 @@ module Parser =
...
+                | x -> Some(System.String.Concat x)
+
+        description <?> "release description body"
+
+    let pRelease: Parser<Release> =
         let vPart = skipString "##" >>. spaces1 >>. pVersion
         let middle = spaces1 .>> pchar '-' .>> spaces1
-        let date = pDate .>> skipRestOfLine true
+        let date = pDate
+        let yanked = opt spaces1 >>. pYanked
```
Comment (MangelMaxime) [inline src/Ionide.KeepAChangelog/Library.fs:253]: I was not able to keep the `.>> skipRestOfLine true` for some reason it was making the tests fails.

**Comment by nojaf** [comment]
If a code repository is not set up to use auto-formatting, you should not use a formatter in your PR!
I tried to address this in https://github.com/ionide/KeepAChangelog/pull/24, but as long as the current code is not being checked on every commit you should not use a formatter.

This would have saved you a lot of effort, to be honest 😸

**Comment by MangelMaxime** [comment]
Well the repository is setup to use fantomas…

It had it installed and asked VSCode to format on save. I have no problem submitting code without formatting the code but then we need to uninstall fantomas and disable the auto-formatting on save configuration.

**Comment by MangelMaxime** [comment]
I am closing because I think going in the direction from #17 makes more sense.

Out sourcing the parser works would make it easier to maintain Ionide.KeepAChangelog.

**Pull request closed without merging**